packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -764,7 +764,27 @@ export class Project extends ProjectBase {
 
         // save SecretsManager env variables
         if (envVariable.type === BuildEnvironmentVariableType.SECRETS_MANAGER) {
-          if (Token.isUnresolved(envVariableValue)) {
+          if (envVariable.value instanceof Object) {
+            const secretArn = (envVariable.value as secretsmanager.Secret).secretArn;
+            const parsedSecretArn = stack.parseArn(secretArn);
+
+            secretsManagerIamResources.add(`${secretArn}*`);
+            if (parsedSecretArn && parsedSecretArn.account &&
+              Token.compareStrings(stack.account, parsedSecretArn.account) === TokenComparison.DIFFERENT) {
```

Comment:
Let's compare against the account of t he  `secret` we created above directly:

```suggestion
            if (Token.compareStrings(stack.account, secret.env.account) === TokenComparison.DIFFERENT) {
```

Reply:
Not sure if this works. It looks like the account under `env` isn't parsed from the Arn. At least the current test breaks for that. I will look into it 👍🏿

Reply:
Sorry? You lost me here 😜.

Reply:
So this is about the change you proposed. If we would do
``` ts
if (Token.compareStrings(stack.account, secret.env.account) === TokenComparison.DIFFERENT)
```
The account in `env` would have to be the one where the secret lives and could be different from the stack account. When creating a secret using `fromSecretCompleteArn` the `env` variables are not parsed from the Arn rather from the stack where we called this. So the comparison won't work

Reply:
Ah yes. Then absolutely, this should be fixed in SecretsManager. Probably something like this: https://github.com/aws/aws-cdk/pull/13708, just for `Secret`, will be good enough.

Reply:
As for this being another PR: I don't want to torture you too much, this is already quite a complex change, so I'm fine with including it as part of this PR, if it makes things easier for you 🙂.

Reply:
I think only changing should suffice here, since Secrets from other accounts can only be imported through `fromSecretAttributes`.
https://github.com/aws/aws-cdk/blob/7d2684edc67b6fc4ea957e1a2a44d726c08e4047/packages/%40aws-cdk/aws-secretsmanager/lib/secret.ts#L422-L428